### PR TITLE
chore: fixing "Cannot create an instance of an interface" on image build

### PR DESCRIPTION
**Pull request by ja0b**
**What this PR does / why it's submitted / why we need it**:
When using the image builder this error was being thrown "MissingMethodException: Cannot create an instance of an interface" this was the best way around it.

Now if someone wants to change the default breakpoints the need to pass this new object with the breakpoint values.

A Dictionary type worked but it couldn't be used from the razor view, somehow the model binder was picking all the properties sent by the action in this dictionary.

---

**Checklist**:

- [x] My PR has a descriptive title (not a vague title like `Needed this`)
- [x] Title and all the commits follow [commitlint guidelines](https://github.com/conventional-changelog/commitlint#what-is-commitlint)
- [x] My PR targets the `develop` branch or appropriate `release` branch

### CODE REVIEW

**Review by imranhaidercogworks — APPROVED** [review]

**Review by lkoruba — APPROVED** [review]

File: Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections.Generic;
+
+namespace Cogworks.Umbraco.Essentials.Models
+{
+    public class ImageBreakpoints
```
Comment (aochmann) [inline Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs:5]: What about creating dedicated class that simplify usage:

1) https://dotnetfiddle.net/3mG4Bd
2) https://dotnetfiddle.net/IZbEjm

**Review by trt — APPROVED** [review]

File: Source/Cogworks.Umbraco.Essentials/Builders/ResponsiveImageBuilder.cs [hunk 2]
```diff
@@ -55,32 +55,40 @@ public class ResponsiveImageBuilder : IResponsiveImageBuilder
             int? quality = null,
             string fileExtension = null)
         {
-            breakpoints ??= BreakPointConstants.DefaultBreakpoints;
+            if (breakpoints.HasValue() && !breakpoints.BreakpointsValues.HasAny())
```
Comment (mzajkowski) [inline Source/Cogworks.Umbraco.Essentials/Builders/ResponsiveImageBuilder.cs:58]: Is logic correct here? We have value (we're not null) AND there is no values added to the BreakpointValues collection? 🤔

File: Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs (hunk 1 above)
Comment (mzajkowski) [reply Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs:5]: I like this idea! Both solutions will work, but here we'll have less coupling and dependencies when we'll be dealing directly on the object/class. Definitely worth considering.

File: Source/Cogworks.Umbraco.Essentials/Builders/ResponsiveImageBuilder.cs (hunk 2 above)
Comment (ja0b) [reply Source/Cogworks.Umbraco.Essentials/Builders/ResponsiveImageBuilder.cs:58]: This is just to make sure the initial object is not null

File: Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs (hunk 1 above)
Comment (ja0b) [reply Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs:5]: I did try this today and I honestly I don't know/see how this is can be simpler/better than a dumb class as the one that is in place, could any of you jump in a call with me and show me your idea. I can merge this as I got the 2 green ticks but I'm curious. please let me know asap as I do want to merge this PR this week. 👍

Comment (aochmann) [reply Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs:5]: Lets wait for tomorrow

Comment (aochmann) [reply Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs:5]: Its just for how to work with ImageBreakpoints, this class will behaves like object with configurations (we will have options - depends on approach we take - to pass list/dictionary of breakpoints or add it via method). From user perspective will be easily to know how to work with it.

For example:

```csharp
var imageBreakpoints = new ImageBreakpoints(new Dictionary<string, string>() { ["key"] = "value" });

imageBreakpoints.AddBreakpoint("key2", "value2"); //depends if we want to give abilities for extending

foreach(var imageBreakpoint in imageBreakpoints.GetBreakpoints())
{
    // logic
}

// or

foreach(var imageBreakpoint in imageBreakpoints)
{
    // logic
} 

if (imageBreakpoints.HasAny())
{
  // logic
}
```

Personal I would go with immutable approach for assigning object with list/dictionary without ability to add/remove new entries.

The constructor of this object will ensure on user to pass list/dictionary of configuration.

Comment (aochmann) [reply Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs:5]: Sorry this method `GetBreakpoints() should be:

```csharp
public IEnumerable<KeyValuePair<string, string>> GetBreakpoints()
        => _breakpoints.ToArray();
```

Comment (aochmann) [reply Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs:5]: Full example: https://dotnetfiddle.net/Dzca6E

Comment (aochmann) [reply Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs:5]: Another approach could be domain crops, example here: https://dotnetfiddle.net/qkscUW but this one is not generic for all.

Comment (aochmann) [reply Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs]: I will create separate PR for the last domain example usage.

File: Source/Cogworks.Umbraco.Essentials/Builders/ResponsiveImageBuilder.cs [hunk 3]
```diff
@@ -55,17 +56,14 @@ public class ResponsiveImageBuilder : IResponsiveImageBuilder
...
             string fileExtension = null)
         {
-            if (breakpoints.HasValue() && !breakpoints.BreakpointsValues.HasAny())
+            if (!breakpoints.HasAny())
             {
-                breakpoints = new ImageBreakpoints
-                {
-                    BreakpointsValues = BreakPointConstants.DefaultBreakpoints
-                };
+                breakpoints = new ImageBreakpoints(BreakPointConstants.DefaultBreakpoints.ToList());
```
Comment (aochmann) [inline Source/Cogworks.Umbraco.Essentials/Builders/ResponsiveImageBuilder.cs:61]: You don't need perform `ToList()`

Comment (ja0b) [reply Source/Cogworks.Umbraco.Essentials/Builders/ResponsiveImageBuilder.cs]: changed 👍

**Review by aochmann — APPROVED** [review]

**Pull request merged**